src/Cake/Program.cs

```diff
@@ -76,6 +76,18 @@ private static IContainer CreateContainer(bool mono)
             builder.RegisterType<WindowsRegistry>().As<IRegistry>().SingleInstance();
             builder.RegisterType<CakeContext>().As<ICakeContext>().SingleInstance();
 
+            // NuGet addins support
+            builder.RegisterType<NuGetVersionUtilityAdapter>()
+                .As<INuGetFrameworkCompatibilityFilter>()
```

Comment:
That would only allow it to be resolved as `NugetVersionUtilityAdapter` instead of as its interface.  I *think* what you're intending to suggest is `.AsImplementedInterfaces().SingleInstance()`.